### ISSUE TITLE
Change maxPlayers to 2

### ISSUE THREAD

**Issue by celadaris** [BeardedManStudios/ForgeNetworkingRemastered#211]
### Version Number and Operating System(s):
Tested on PC - Windows 10
Forge Version - 20.12.2018
Unity - 2018.2.13f1

### Expected behavior:
Change the max amount of players to 2 (host and client), this should block all other connections.

### Actual behavior:
2nd client hijacks the connection and is able to receive updates (for a couple seconds) from the network while the original client isn't. You can see this after connecting the second client and then quickly interacting with the host. Both clients are unable to receive updates shortly after.

Adding a 3rd or more clients will have them hijack the network to receive a single (sometimes more) update(s) to the networkObject.

If all clients that are actually connected completely exit then you can connect another client and it will resume as normal, but once again adding more clients will commence the hijacking.

### Steps to reproduce:
You can check out my example [here](https://github.com/celadaris/maxPlayers-bug)
Change MultiplayerMenu.cs from...
```cs
else
       {
	server = new UDPServer(64);

	if (natServerHost.Trim().Length == 0)
            ((UDPServer)server).Connect(ipAddress.text, ushort.Parse(portNumber.text));
	else
	    ((UDPServer)server).Connect(port: ushort.Parse(portNumber.text), natHost: natServerHost, natPort: natServerPort);

      }
```
to
```cs
else
       {
        int maxAllowedPlayers = 2;
        // set maxPlayers to 2
        server = new UDPServer(maxAllowedPlayers);

        if (natServerHost.Trim().Length == 0)
            ((UDPServer)server).Connect(ipAddress.text, ushort.Parse(portNumber.text));
        else
            ((UDPServer)server).Connect(port: ushort.Parse(portNumber.text), natHost: natServerHost, natPort: natServerPort);

       }
```

#### [Optional] Discord Username:
.̴̓̈ͤ͠҉͜~̆̃͋̉ͪ̇Celadaris#9541

**Comment by phalasz** [comment]
I did some debugging of this with the latest github repo files.

This is what is going on. Host starts up. Client 1 connects. Client 2 tries to connect, but gets rejected by the server so it retries the connection 10 times as that is the max retry count. After that client 1 times out and client 2 can connect.

Client 1 seems broken, but that is only because it is not going back to the menu scene on timeout.

In conclusion the client should stop trying to connect if it receives a max connection error message from the server. Also it is a bit worrying that an already connected client times out right when the other client reaches the retry count.

**Comment by phalasz** [comment]
Also this issue looks a lot like issue #141

**Comment by phalasz** [comment]
Some more digging shows that the problem seems to be with

```
if (Players.Count == MaxConnections)
			{
				// Tell the client why they are being disconnected
				Send(Error.CreateErrorMessage(Time.Timestep, "Max Players Reached On Server", false, MessageGroupIds.MAX_CONNECTIONS, true));

				// Send the close connection frame to the client
				Send(new ConnectionClose(Time.Timestep, false, Receivers.Target, MessageGroupIds.DISCONNECT, false));

				return;
}
```

The `Send` in the `UDPServer::SetupClient` method only sends to accepted clients, which means that it will send the error message and the connection close message to all connected clients that are set to be Receivers by `CommonServer::PlayerIsReceiver`. This causes all connected and accepted clients to disconnect instead of the not accepted client getting the error message.

**Comment by phalasz** [comment]
@celadaris would you be able to validate that the fix we merged solves your problem?

**Comment by phalasz** [comment]
Can someone with the powers close this issue? There were no more reports of this and the ticket has been open for quite some time now with a fix merged.

**Comment by phalasz** [comment]
No activity on this ticket since the fix, so this can be closed now.
